### PR TITLE
WIP Update dataproc package

### PR DESCRIPTION
**Pull request by michalslowikowski00**
Update Dataproc package to 2.2.0. In progress.

I followed the UPGRADING.md
https://github.com/googleapis/python-dataproc/blob/master/UPGRADING.md. 
As you can see they recommend to use script `fixup_dataproc_v1_keywords.py`. 
But not necessarily. I think. I have done it, but after that system test doesn't work.
I removed suggested param:

```
request={
    'project_id' : "project_id", 'region' : "region"}

```
and leave code as is, but I had to remove param `request_id=request_id`. 
That was the Tomek's fix for idempotency.
New api doesn't support it. Sorry. :(

New API is represented by this:

```
def get_cluster(
        self,
        # the new parameter
        request: clusters.GetClusterRequest = None,
        # the new parameter
        *,
        # the old ones
        project_id: str = None,
        region: str = None,
        cluster_name: str = None,
        retry: retries.Retry = gapic_v1.method.DEFAULT,
        timeout: float = None,
        metadata: Sequence[Tuple[str, str]] = (),
    ) -> clusters.Cluster:
```

New params are `request: clusters.GetClusterRequest` and `*`.
But after use this `fixup_dataproc_v1_keywords.py` script new key arg has been added to the methods and system test doesn't work so I removed new param `request={}`. 
I had to remove also `request_id=request_id`, `cluster_uuid: Optional[str]` and `graceful_decommission_timeout`. 
The job states also have been changed. from `JobStatus.ERROR` to `JobStatus.State.ERROR`.

The question is how we want to refactor the dataproc, because both of these calls are valid:

The new approch suggested by Google:
```
response = client.get_cluster(
    request={
        "project_id": project_id,
        "region": region,
        "cluster_name": cluster_name
    }
)
```
The old one, with changes (removed request_id, luster_uuid: Optional[str] and graceful_decommission_timeout)
```
response = client.get_cluster(
    project_id=project_id,
    region=region,
    cluster_name=cluster_name
)
```
Now system test is green, but is it satisfying for us?
WDYT?

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/new_dataproc.py [hunk 1]
```diff
@@ -0,0 +1,892 @@
...
+        # [a-z]([-a-z0-9]*[a-z0-9])? (current airflow version string follows
+        # semantic versioning spec: x.y.z).
+        labels = labels or {}
+        labels.update({'airflow-version': 'v' + airflow_version.replace('.', '-').replace('+', '-')})
+
+        cluster = {
+            "project_id": project_id,
+            "cluster_name": cluster_name,
+            "config": cluster_config,
+            "labels": labels,
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/new_dataproc.py:286]:
```suggestion
            "labels": labels,
            "request_id": request_id,
```

File: airflow/providers/google/cloud/hooks/new_dataproc.py [hunk 2]
```diff
@@ -0,0 +1,892 @@
...
+    WorkflowTemplateServiceClient,
+)
+from google.protobuf.internal.well_known_types import FieldMask
+
+from airflow.exceptions import AirflowException
+from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
+from airflow.version import version as airflow_version
+
+
+class DataProcJobBuilder:
```
Comment (turbaszek) [inline airflow/providers/google/cloud/hooks/new_dataproc.py:47]: Do we need job builder? It's only required for deprecated operators

**Comment by turbaszek** [comment]
Imho we should build requests for users. This should also allow us preserve the backward compatibility. 

In case of the PR - would you mind to reduce the number of changes? I'm not sure what parts are relevant and should be reviewed and what should be skipped

File: airflow/providers/google/cloud/hooks/new_dataproc.py (hunk 1 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/new_dataproc.py:286]: This param causes exception.
```
Traceback (most recent call last):
  File "/opt/airflow/airflow/models/taskinstance.py", line 1086, in _run_raw_task
    self._prepare_and_execute_task_with_callbacks(context, task)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1260, in _prepare_and_execute_task_with_callbacks
    result = self._execute_task(context, task_copy)
  File "/opt/airflow/airflow/models/taskinstance.py", line 1300, in _execute_task
    result = task_copy.execute(context=context)
  File "/opt/airflow/airflow/providers/google/cloud/operators/dataproc.py", line 605, in execute
    cluster = self._create_cluster(hook)
  File "/opt/airflow/airflow/providers/google/cloud/operators/dataproc.py", line 541, in _create_cluster
    metadata=self.metadata,
  File "/opt/airflow/airflow/providers/google/common/hooks/base_google.py", line 383, in inner_wrapper
    return func(self, *args, **kwargs)
  File "/opt/airflow/airflow/providers/google/cloud/hooks/new_dataproc.py", line 298, in create_cluster
    metadata=metadata,
  File "/usr/local/lib/python3.7/site-packages/google/cloud/dataproc_v1/services/cluster_controller/client.py", line 406, in create_cluster
    request.cluster = cluster
  File "/usr/local/lib/python3.7/site-packages/proto/message.py", line 579, in __setattr__
    pb_value = marshal.to_proto(pb_type, value)
  File "/usr/local/lib/python3.7/site-packages/proto/marshal/marshal.py", line 205, in to_proto
    pb_value = rule.to_proto(value)
  File "/usr/local/lib/python3.7/site-packages/proto/marshal/rules/message.py", line 32, in to_proto
    return self._descriptor(**value)
ValueError: Protocol message Cluster has no "request_id" field.
```

File: airflow/providers/google/cloud/hooks/new_dataproc.py (hunk 2 above)
Comment (michalslowikowski00) [reply airflow/providers/google/cloud/hooks/new_dataproc.py:47]: I do not know, I can delete it.

File: airflow/providers/google/cloud/hooks/new_dataproc.py (hunk 1 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/new_dataproc.py:286]: There is `request` parameter which allows to pass `SubmitJobRequest` which allows users to pass `request_id`

Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/new_dataproc.py:286]: Also there's a new tooling around this `SubmitJobRequest` which may allow us to improve the idempotency

**Pull request closed without merging**